Provis.Core/Services/WorkspaceService.cs

```diff
@@ -265,5 +266,38 @@ public async Task<WorkspaceInfoDTO> GetWorkspaceInfoAsync(int workspId, string u
 
             return workspace;
         }
+
+        public async Task DeleteFromWorkspaceAsync(int workspaceId, DeleteUserDTO deleteUserDTO)
+        {
+            var user = await _userManager.FindByEmailAsync(deleteUserDTO.UserEmail);
+
+            if (user == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound, 
+                    "User with this Id not exist");
+            }
+
+            var workspace = _workspaceRepository.GetByKeyAsync(workspaceId);
+
+            if (workspace == null)
+            {
+                throw new HttpException(System.Net.HttpStatusCode.NotFound, 
+                    "Workspace with Id not exist");
+            }
+
+            var userWorksp = _userWorkspaceRepository
+                .Query()
+                .Where(x => x.WorkspaceId == workspaceId && x.User.Email == deleteUserDTO.UserEmail)
```

Comment:
Please optimaze the request to db, look on execution plan and make sure that indexes seek is used for it